### PR TITLE
feature(time): implement start property for profiling end and start times of messages

### PR DESCRIPTION
**Pull request by iammerrick**
# Work in Progress
- [x] Implement API
- [ ] Implement ConsoleReporter to support timestamps
- [ ] Code Review

This branch is the beginnings of profiling...

``` javascript
let end = logger.start.info('start some pretty hardcore work');
// All this crazy work.
end('woah, that was tiring.');
```

This would log out two messages from core: 

``` javascript
// On first call
{
  level: 'info',
  group: 'feature',
  message: 'start some pretty hardcore work',
  timestamp: 1395002162444,
  guid: 1
}

// on end call
{
  level: 'info',
  group: 'feature',
  message: 'woah, that was tiring.',
  timestamp: 1395002162448,
  guid: 2,
  endOf: 1
}
```

[Review on Reviewable](https://reviewable.io/reviews/angular/diary.js/6)

### CODE REVIEW

**Comment by iammerrick** [comment]
I ran this by my friend @peanav and he noted that:

``` javascript
let end = logger.info.start('start some pretty hardcore work');
```

Would be more readable, I agree with him. What are your thoughts on that API @IgorMinar?

**Comment by IgorMinar** [comment]
at that point then how about

``` js
let end = logger.infoStart('start some pretty hardcore work');
```

**Comment by IgorMinar** [comment]
shouldn't the first event contain some flag that would say that this is just the start event and there is more to come?

File: test/diary.spec.js [hunk 1]
```diff
@@ -117,7 +119,46 @@ describe('Diary', () => {
     httpLogger.info('hello');
     logger.info('global');
     expect(http.receive.calls.count()).toEqual(1);
-    expect(http.receive).toHaveBeenCalledWith({ level: 'info', group: 'http', message: 'hello' });
+    expect(http.receive).toHaveBeenCalledWith(jasmine.objectContaining({ level: 'info', group: 'http', message: 'hello' }));
+
+  });
+
+  describe('time', () => {
```
Comment (IgorMinar) [inline test/diary.spec.js:126]: `'start'` ?

**Comment by IgorMinar** [comment]
I like this implementation :)

**Pull request closed without merging**